### ISSUE TITLE
`bundle install` will install `Bundler version 2.0.2` despite being unsupported by cocoapods.gemspec

### ISSUE THREAD

**Issue by Coeur** [CocoaPods/CocoaPods#8920]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

Steps:
1. clone CocoaPods repository
2. delete `Gemfile.lock` (OK, normally you don't, but I was trying [this workaround](https://github.com/bundler/bundler/issues/6882#issuecomment-451739257))
3. be on bundler version 1.x (`gem install bundler -v '1.17.3'`)
4. run `bundle install` and be surprised that it will install Bundler version 2.0.2 and fail to meet the requirement of ` s.add_development_dependency 'bundler', '~> 1.3'` from `cocoapods.gemspec`

```bash
$ bundle --version
Bundler version 1.17.3
$ bundle install
Fetching https://github.com/segiddins/json.git
Fetching https://github.com/CocoaPods/CLAide.git
Fetching https://github.com/CocoaPods/Core.git
Fetching https://github.com/CocoaPods/cocoapods-deintegrate.git
Fetching https://github.com/CocoaPods/cocoapods-downloader.git
Fetching https://github.com/CocoaPods/cocoapods-plugins.git
Fetching https://github.com/CocoaPods/cocoapods-search.git
Fetching https://github.com/CocoaPods/cocoapods-stats.git
Fetching https://github.com/CocoaPods/cocoapods-trunk.git
Fetching https://github.com/CocoaPods/cocoapods-try.git
Fetching https://github.com/CocoaPods/Molinillo.git
Fetching https://github.com/CocoaPods/Nanaimo.git
Fetching https://github.com/CocoaPods/Xcodeproj.git
Fetching gem metadata from https://rubygems.org/..........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    bundler (~> 1.3)

  Current Bundler version:
    bundler (2.0.2)
This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?

Could not find gem 'bundler (~> 1.3)' in any of the relevant sources:
  the local ruby installation
$ bundle --version
Bundler version 2.0.2
```

I'm not sure why the above is happening, so I workaround it locally by editing `cocoapods.gemspec` like that:
```ruby
-  s.add_development_dependency 'bundler', '~> 1.3'
+  s.add_development_dependency 'bundler', '>= 1.3'
```

**Comment by dnkoutso** [comment]
Is it because its a development dependency? I am not sure how those two are connected.

**Comment by Coeur** [comment]
@dnkoutso I don't know either. Would it be an issue to change it to `'>= 1.3'` and allow Bundler 2.0.2 instead of forbidding it?

**Comment by dnkoutso** [comment]
If you do `>= 1.3', '< 2.0'` does it exhibit same problem?

**Comment by Coeur** [comment]
OK, I waste too much time on it:
>[Bundler 2 introduced a new feature that will automatically switch between Bundler v1 and v2 based on the lockfile.](https://bundler.io/blog/2019/01/04/an-update-on-the-bundler-2-release.html)

Which means that when someone has this output:
```bash
$ bundle --version
Bundler version 1.17.3
```
... it doesn't really tell if you have bundler 1.17.3 or 2.0.2! You need to change to a different folder or rename `Gemfile.lock` temporary to get to know the actual version of your installation! This has some other nasty side-effects, like triggering `GemNotFoundException` when you only want to check the installed version:
```bash
$ sudo gem uninstall bundler
[..]
$ sudo gem install bundler
Fetching bundler-2.0.2.gem
Successfully installed bundler-2.0.2
1 gem installed
$ bundle --version
/Library/Ruby/Site/2.3.0/rubygems.rb:284:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /Redacted/CocoaPods/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:1.17.3`
	from /Library/Ruby/Site/2.3.0/rubygems.rb:303:in `activate_bin_path'
	from /usr/local/bin/bundle:23:in `<main>'
```

So imagine that you have both versions of bundler installed (1.17.3 and 2.0.2), you'll get the error message from my first message.
>If you do '>= 1.3', '< 2.0' does it exhibit same problem?
Yes, same issue. One can only `bundle install` with a missing Gemfile.lock by uninstalling bundler 2.0.2. Or by allowing bundler 2.x in the `cocoapods.gemspec`.

**Comment by dnkoutso** [comment]
Is this something to be fixed here then?

**Comment by Coeur** [comment]
I'd say that... if I'm the only one noticing the issue, it means that people aren't deleting the `gemfile.lock` very often. So we can close our issue for now and hope for a fix at bundler/bundler#7099.

(the alternative being to change our `cocoapods.gemspec` to have `s.add_development_dependency 'bundler', '>= 1.3', '< 3.0'`, but I don't know if we want to jump to bundler version 2 yet.)

**Comment by dnkoutso** [comment]
OK! we can re-open if more reports come in and people can search for this. Thanks again for the investigation.
